frontend/app_flowy/lib/workspace/presentation/settings/widgets/settings_appearance_view.dart

```diff
@@ -28,6 +30,124 @@ class SettingsAppearanceView extends StatelessWidget {
   }
 }
 
+class ThemeTypeSetting extends StatelessWidget {
+  final ThemeType currentThemeType;
+  const ThemeTypeSetting({required this.currentThemeType, super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      children: [
+        Expanded(
+          child: FlowyText.medium(
+            LocaleKeys.settings_appearance_themeType_label.tr(),
+            overflow: TextOverflow.ellipsis,
+          ),
+        ),
+        AppFlowyPopover(
+          direction: PopoverDirection.bottomWithRightAligned,
+          child: FlowyTextButton(
+            _getThemeName(currentThemeType),
+            fillColor: Colors.transparent,
+            hoverColor: Theme.of(context).colorScheme.secondary,
+            onPressed: () {},
+          ),
+          popupBuilder: (BuildContext context) {
+            return IntrinsicWidth(
+              child: Column(
+                mainAxisSize: MainAxisSize.min,
+                children: [
+                  _themeTypeItemButton(context, ThemeType.official),
+                  _themeTypeItemButton(context, ThemeType.dandelion),
+                ],
+              ),
+            );
+          },
+        ),
+      ],
+    );
+  }
+
+  Widget _themeTypeItemButton(BuildContext context, ThemeType themeType) {
+    return SizedBox(
+      height: 32,
+      child: FlowyButton(
+        text: FlowyText.medium(_getThemeName(themeType)),
+        rightIcon: currentThemeType == themeType
+            ? svgWidget("grid/checkmark")
+            : const SizedBox(),
+        onTap: () {
+          context
+              .read<AppearanceSettingsCubit>()
+              .setTheme(_getThemeName(themeType));
+          if (currentThemeType != themeType) {
+            context.read<AppearanceSettingsCubit>().setThemeType(themeType);
+          }
+        },
+      ),
+    );
+  }
+
+  String _getThemeName(ThemeType themeType) {
+    switch (themeType) {
+      case ThemeType.official:
+        return LocaleKeys.settings_appearance_themeType_defaultTheme.tr();
+      case ThemeType.dandelion:
+        return LocaleKeys.settings_appearance_themeType_dandelionCommunity.tr();
+      default:
+        throw Exception("Unknown ThemeType");
+    }
+  }
+}
+
+// class ThemeNameSetting extends StatelessWidget {
```

Comment:
Please remove those commented codes if they won't be used anymore.

Reply:
okay

Reply:
That depends on the way we want to give choice of selecting themes to the user. The uncommented code is for the consistent popover choice of theme selection and the commented code is for a list view of themes as I have stated in my earlier comments in this PR. So what do you think is better, a popover or a listview. According to your reply, I will have to delete the other one.

Reply:
I prefer to popover style. Or you can add a comment above the code to explain why we should keep them.

Reply:
I too prefer the popover, because it is consistent with the rest of the User Experience on AppFlowy. No problem I have removed the listview comment.